### PR TITLE
link: Add support to configure NIC Ring buffer size

### PR DESCRIPTION
**Pull request by ssahani**
closes https://github.com/systemd/systemd/issues/13601

### CODE REVIEW

File: man/systemd.link.xml [hunk 1]
```diff
@@ -624,6 +624,19 @@
...
+      <varlistentry>
+        <term><varname>RxBufferSize=</varname></term>
+        <listitem>
+          <para>Takes a integer. Specifies the NIC receive ring buffer size. When unset, the kernel's default will be used.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>TxBufferSize=</varname></term>
+        <listitem>
+          <para>Takes a integer. Specifies the NIC transmit ring buffer size. When unset, the kernel's default will be used.Takes a boolean.</para>
```
Comment (yuwata) [inline man/systemd.link.xml]: Please drop `takes a Boolean.`

File: src/shared/ethtool-util.c [hunk 2]
```diff
@@ -858,3 +904,45 @@ int config_parse_advertise(const char *unit,
...
+        assert(data);
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse interface buffer value, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (k < 1) {
+                log_syntax(unit, LOG_ERR, filename, line, -EINVAL, "Invalid %s value, ignoring: %s", lvalue, rvalue);
```
Comment (yuwata) [inline src/shared/ethtool-util.c]: `-EINVAL` -> 0

**Comment by ssahani** [comment]
updated thanks for the review @yuwata

**Comment by yuwata** [comment]
Semaphore CI failure is not related.

**Pull request merged**